action_server/src/action_server/server.py

```diff
@@ -36,48 +36,52 @@ def _get_actions_cb(self, req):
-            feedback.current_subtask = self._task_manager.get_next_action_name()
-            self._action_server.publish_feedback(feedback)
-
-            action_result = self._task_manager.execute_next_action()
-            rospy.logdebug("Result of action execution: {}".format(action_result))
-            self._result.log_messages.append(action_result.message)
-            if not action_result.succeeded:
-                self._result.result = action_server_msgs.msg.TaskResult.RESULT_TASK_EXECUTION_FAILED
+                if configuration_result.missing_field:
+                    self._result.result = action_server_msgs.msg.TaskResult.RESULT_MISSING_INFORMATION
+                    self._result.missing_field = "actions" + configuration_result.missing_field
+                    self._result.log_messages.append(" I don't have enough information to perform that task.")
+                    self._result.log_messages.append(configuration_result.message)
+                elif configuration_result.message:
+                    # TODO: this task result should not be RESULT_UNKNOWN
+                    self._result.result = action_server_msgs.msg.TaskResult.RESULT_UNKNOWN
+                    self._result.log_messages.append(configuration_result.message)
+                else:
+                    self._result.result = action_server_msgs.msg.TaskResult.RESULT_UNKNOWN
+                    self._result.log_messages.append(" It seems that I am unable to perform that task. "
+                                                       "Not sure why though.")
                 self._action_server.set_aborted(self._result)
                 self._task_manager.clear()
-                rospy.logdebug("Execution of state machine aborted because action failed.")
+                rospy.logerr("Setting up state machine failed")
                 return
 
+            while not self._task_manager.done:
+                # Pass feedback to client about what type of action is running
+                feedback = action_server_msgs.msg.TaskFeedback()
+                feedback.current_subtask = self._task_manager.get_next_action_name()
+                self._action_server.publish_feedback(feedback)
+
+                action_result = self._task_manager.execute_next_action()
+                rospy.logdebug("Result of action execution: {}".format(action_result))
+                self._result.log_messages.append(action_result.message)
+                if not action_result.succeeded:
+                    self._result.result = action_server_msgs.msg.TaskResult.RESULT_TASK_EXECUTION_FAILED
+                    self._action_server.set_aborted(self._result)
+                    self._task_manager.clear()
+                    rospy.logdebug("Execution of state machine aborted because action failed.")
+                    return
+        except Exception as e:
+            raise type(e), type(e)("An error occurred using task recipe: %s\n" % goal.recipe + e.message)
```

Comment:
I've never seen this construction before. What's some example output? Does this give you the ninfo needed  with the stacktrace of where things went south?